Add Resolc 0.4.0 compiler

elle-j opened the pull request:
## What

Adds [Revive's Resolc](https://github.com/paritytech/revive) Solidity compiler.

## CE

This infra addition is used by the accompanying CE PR: https://github.com/compiler-explorer/compiler-explorer/pull/8164

mattgodbolt commented:
> The only dependency of the resolc compiler frontend executable is a solc compiler build (any version >=0.8.0, by default it will use the one found in $PATH). Is it fine to have this dependency implicit or should it be expressed explicitly?

It will need to be expressed explicitly. There is no `solc` compiler on any PATH, our systems have no compilers on the path and need explicit references. The "kotlin.yaml" compiler installation is an example of doing this during installation (if that's required); and the kotlin.ts in the main project shows a way to configure which "java" (kotlin's dependency) is used to execute kotlin with.

Let me know if I can help more.

elle-j commented:
Thanks for the reminder! Added the dependency [here](https://github.com/compiler-explorer/infra/pull/1855/commits/36da96163e0687198f79ad463d1934c3ac60b3ca), and updated the [compiler options](https://github.com/compiler-explorer/compiler-explorer/pull/8164/commits/fc37e4465a1e501cce3e79cf7fc850f0acf5a0b2) on the CE side with the solc path which I believe should exist there in production as well right?
